### PR TITLE
Add README for bidirectional sample

### PR DESCRIPTION
**Pull request by zackliu**

### CODE REVIEW

File: samples/bidirectional-chat/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Azure function bidirectional chatroom sample
+
+This sample demonstrate how to integrate SignalR Service upstream feature with Azure Function. This features SignalR Service sending messages to upstream endpoints in serverless scenario. With the signalr function binding, you can leverge class based model in C# that can give you a consistent experience with non-serverless scanarios.
```
Comment (vicancy) [inline samples/bidirectional-chat/README.md:3]: `demonstrate` => `demonstrates`

I feel the first line introduces in new terms such as "upstream", which might confuse people from the very beginning. How about having a simple introduction, that it is a complete fully functional sample that can have chatroom running towards Azure Function?

File: samples/bidirectional-chat/README.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
+# Azure function bidirectional chatroom sample
+
+This is a chatroom sample that demonstrates bidirectional message pushing between Azure SignalR Service and Azure Function in serverless scenario. It leverage the "upstream" provided by Azure SignalR Service that features proxying messages from client to upstream endpoints in serverless scenario. Azure Functions with signalr trigger binding allow you to write code to receive and push messages in several languages, including JavaScript, Python, C#, etc.
```
Comment (JialinXin) [inline samples/bidirectional-chat/README.md:3]: >leverage [](start = 150, length = 8)

leverages

Comment (JialinXin) [inline samples/bidirectional-chat/README.md:3]: >upstream [](start = 164, length = 8)

strong "upstream"? do we have any document for detail explanation about upstream that can be link here?

Comment (JialinXin) [inline samples/bidirectional-chat/README.md:3]: >signalr [](start = 319, length = 7)

SignalR

File: samples/bidirectional-chat/README.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+The following softwares are required to build this tutorial.
+* [.NET SDK](https://dotnet.microsoft.com/download) (Version 3.1, required for Functions extensions)
+* [Azure Functions Core Tools](https://docs.microsoft.com/en-us/azure/azure-functions/functions-run-local?tabs=windows%2Ccsharp%2Cbash#install-the-azure-functions-core-tools) (Version 3)
+* [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest)
+
+<a name="run-sample-in-azure"></a>
+
+## Run sample in Azure
+
+It's a quick try of this sample. You will create an Azure Signalr Service and an Azure Function app to host sample. And you will launch chatroom locally but connecting to Azure SignalR Service and Azure Function.
```
Comment (JialinXin) [inline samples/bidirectional-chat/README.md:21]: >Signalr [](start = 58, length = 7)

SignalR

File: samples/bidirectional-chat/README.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+### Create SignalR Service
```
Comment (JialinXin) [inline samples/bidirectional-chat/README.md:23]: >SignalR Service [](start = 11, length = 15)

Azure SignalR Service

File: samples/bidirectional-chat/README.md (hunk 2 above)
Comment (zackliu) [reply samples/bidirectional-chat/README.md:3]: Azure Functions is plural so it should be `allow`?

File: samples/bidirectional-chat/README.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+2. Update application settings
+
+    ```bash
+    az functionapp config appsettings set --resource-group <resource_group_name> --name <function_name> --setting AzureSignalRConnectionString="<signalr_connection_string>"
+    ```
+
+3. Update Azure SignalR upstream settings
+
+    Open the Azure Portal and nevigate to the Function App created before. Find `signalr_extension` key in the App keys blade.
```
Comment (JialinXin) [inline samples/bidirectional-chat/README.md:83]: >App keys [](start = 111, length = 8)

strong App keys and later CORS blade to highlight.

File: samples/bidirectional-chat/README.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    az resource update --ids <signalr-resource-id> --set properties.upstream.templates="[{'UrlTemplate': '<function-url>/runtime/webhooks/signalr?code=<signalr_extension-key>', 'EventPattern': '*', 'HubPattern': '*', 'CategoryPattern': '*'}]"
+    ```
+
+### Use a chat sample website to test end to end
+
+1. Enable function app cross origin resource sharing (CORS)
+
+    Although there is a CORS setting in local.settings.json, it is not propagated to the function app in Azure. You need to set it separately.
+
+    1. Open the function app in the Azure portal.
```
Comment (JialinXin) [inline samples/bidirectional-chat/README.md:99]: >portal [](start = 42, length = 6)

Portal ?

File: samples/bidirectional-chat/README.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+2. Install [Live Server](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer) for your VS Code, that can serve web pages locally
+3. Open `bidirectional-chat/content/index.html` and edit base url
+
+    ```js
+    window.apiBaseUrl = '<function-app-url>';
+    ```
+
+4. With **index.html** open, start Live Server by opening the VS Code command palette (F1) and selecting **Live Server: Open with Live Server**. Live Server will open the application in a browser.
+
+5. Try send messages by entering them into the main chat box.
```
Comment (JialinXin) [inline samples/bidirectional-chat/README.md:115]: Maybe add a snapshot about the chatroom sample view in the front or end is more intuitive.

File: samples/bidirectional-chat/README.md (hunk 2 above)
Comment (zackliu) [reply samples/bidirectional-chat/README.md:3]: We don't have public document about `upstream` now. We can update it when we have it.

Comment (JialinXin) [reply samples/bidirectional-chat/README.md]: yeah, then update SignalR is enough.

---
In reply to: [422756824](https://github.com/Azure/azure-functions-signalrservice-extension/pull/120#discussion_r422756824) [](ancestors = 422756824)

**Pull request merged**